### PR TITLE
allow non-utf8 arguments to proxies

### PR DESCRIPTION
**Pull request by euclio**
Fixes rust-lang/rust#56146.

### CODE REVIEW

File: src/rustup-cli/proxy_mode.rs [hunk 1]
```diff
@@ -26,14 +26,15 @@ pub fn main() -> Result<()> {
...
-
+        let toolchain = arg1.as_ref()
+            .and_then(|arg1| arg1.to_str())
+            .and_then(|arg1| {
+                if arg1.starts_with('+') {
+                    Some(&arg1[1..])
+                } else {
+                    None
+                }
+            });
```
Comment (kinnison) [inline src/rustup-cli/proxy_mode.rs]: Could we do the `.starts_with('+')` before the `.to_str()` and issue an error if the user has specified a non-utf8 toolchain name?

**Review by kinnison — COMMENTED** [review]
Other than being unsure what 'proxies' are in this context, I think this change looks sensible.  Certainly preferable to a panic, though I worry if the user might be surprised if they enter a non-utf8 toolchain name and are then confused when instead of honouring it, it is silently ignored.

**Comment by nrc** [comment]
This looks good, but I agree it would be better to warn the user rather than silently ignore it.

File: src/rustup-cli/proxy_mode.rs (hunk 1 above)
Comment (euclio) [reply src/rustup-cli/proxy_mode.rs]: Added the check and two more tests.

**Review by kinnison — APPROVED** [review]
This seems to do as requested, thank you for the change and the tests.

**Comment by nrc** [comment]
Thanks!

**Pull request merged**